components/dashboard/src/components/Modal.tsx

```diff
@@ -150,7 +150,7 @@ export const ModalFooter: FC<ModalFooterProps> = ({ error, warning, children })
     return (
         <div className="relative">
             {hasAlert && (
-                <div className="absolute -top-12 left-0 right-0" style={alertStyles}>
+                <div className="absolute bottom-12 left-0 right-0" style={alertStyles}>
```

Comment:
**suggestion**: Since we're having validation inline for text input, we could do the following in this or a follow PR:
1. Skip the alert here and rely on the SSO clients list to communicate an error or a failed attempt.
2. Migrate away from the modals to a full page layout as we do with the access tokens flow, which is more appropriate, scalable, and accessible.

| Input validation | Error feedback through list | Error feedback in full page |
|-|-|-|
| <img width="552" alt="Screenshot 2023-03-16 at 14 36 12" src="https://user-images.githubusercontent.com/120486/225619354-326b9e3b-d0a0-4d5e-870c-5b60e57cb5e8.png"> | <img width="890" alt="Screenshot 2023-03-16 at 14 36 46" src="https://user-images.githubusercontent.com/120486/225619442-d08678c0-8d4e-4f46-b531-b49c3ea1432f.png"> | <img width="616" alt="Screenshot 2023-03-16 at 14 36 58" src="https://user-images.githubusercontent.com/120486/225619552-ca0a5a71-1d17-4de8-baea-9e5ff3539be1.png"> |

Reply:
Good points. 

If we stuck with the modal, we'd need some version of an alert for any error messages returned from the api call during create/update.

I'd be on board with using full pages here. I debated going down that route, but felt like a bigger change I didn't want to introduce quite yet. It would also simplify some of the complexity in the git auth modal that deals with managing state after creation.

Either way, I think having a way to show simple error messages for a modal in a consistent way will helpful, but need to address some of the usability points we've discussed.

Reply:
![over-here](https://user-images.githubusercontent.com/120486/225753552-3c5b6ed8-71c5-4bb2-b7af-b806e1ac461c.gif)